src/components/AppHeader/AppHeader.tsx

```diff
@@ -24,6 +31,57 @@ const AppHeader = () => {
     const { data: activeAccount } = api.account.useActiveAccount();
     const { localize } = useTranslations();
     const oauthUrl = getOauthUrl();
+    const [isOAuth2Enabled] = useOAuth2Enabled();
+
+    const WSLogoutAndRedirect = async () => {
+        await logout();
+        redirectToOauth();
+    };
+
+    useEffect(() => {
+        const onMessage = async (event: MessageEvent) => {
+            const allowedOrigin = getOAuthOrigin();
+            if (allowedOrigin === event.origin) {
+                if (event.data === 'logout_complete') {
+                    WSLogoutAndRedirect();
+                } else {
+                    // eslint-disable-next-line no-console
+                    console.warn('Unexpected message received: ', event.data);
+                }
+            } else {
+                // eslint-disable-next-line no-console
+                console.warn('Unexpected postmessage origin: ', event.origin);
+            }
+        };
+
+        window.addEventListener('message', onMessage);
+
+        return () => {
+            window.removeEventListener('message', onMessage);
+        };
+    }, []);
+
+    const handleLogout = async () => {
+        if (isOAuth2Enabled) {
+            let iframe: HTMLIFrameElement | null = document.getElementById('logout-iframe') as HTMLIFrameElement;
+            if (!iframe) {
+                iframe = document.createElement('iframe');
+                iframe.id = 'logout-iframe';
+                iframe.style.display = 'none';
+                document.body.appendChild(iframe);
+            }
+
+            iframe.src = getOAuthLogoutUrl();
```

Comment:
will the user be also logged out in other apps?

Reply:
yes, logged out in other apps will be handled by oauth BE service, so we don't need to cater for it in FE